handler/handler.go

```diff
@@ -0,0 +1,121 @@
+package handler
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"runtime"
+	"time"
+
+	nctx "github.com/acaloiaro/neoq/context"
+)
+
+const (
+	DefaultHandlerDeadline = 30 * time.Second
+)
+
+// HandlerFunc is a function that Handlers execute for every Job on a queue
+type HandlerFunc func(ctx context.Context) error
+
+// Handler handles jobs on a queue
+type Handler struct {
+	Handle        HandlerFunc
+	Concurrency   int
+	Deadline      time.Duration
+	QueueCapacity int64
+}
+
+// HandlerOption is function that sets optional configuration for Handlers
+type HandlerOption func(w *Handler)
+
+// WithOptions sets one or more options on handler
+func (h *Handler) WithOptions(opts ...HandlerOption) {
+	for _, opt := range opts {
+		opt(h)
+	}
+}
+
+// HandlerDeadline configures handlers with a time deadline for every executed job
+// The deadline is the amount of time that can be spent executing the handler's HandlerFunc
+// when a deadline is exceeded, the job is failed and enters its retry phase
+func HandlerDeadline(d time.Duration) HandlerOption {
+	return func(h *Handler) {
+		h.Deadline = d
+	}
+}
+
+// HandlerConcurrency configures Neoq handlers to process jobs concurrently
+// the default concurrency is the number of (v)CPUs on the machine running Neoq
+func HandlerConcurrency(c int) HandlerOption {
```

Comment:
Since the package is named handler, we can drop the prefix from all these funcs.